### DISCUSSION TITLE
Doc request: why Remirror over just using ProseMirror directly?

### DISCUSSION THREAD

**Discussion by scottmessinger** [remirror/remirror#568] [General] [answered]
Hi! This project looks great! It looks ambitious and convenient. However, in reading the docs, I was left wondering -- what advantage is there over just using ProseMirror directly, especially if you're not using the React component? Point me to the place in the docs where this is if I missed it!

Obviously, this library adds a number of presets and makes things easy to set up and that's all great. My understanding is it's relatively easy to add plugins/schemas to ProseMirror and, if a company is making a bet on using PM, it wouldn't be that much work (relatively) to just set it up themselves. I could be totally wrong about this -- hence the question! Assuming the main value prop of this library is it saves me time to set up, I was looking for a bit in the docs that works to quantify that time to give a sense of how useful this is. Something like, "Without Remirror, setting up PM requires <X steps>. With Remirror, it takes <steps>" and a list of all the steps Remirror saves. Also, I'd love to see a place that says, "Downsides to using Remirror instead of PM directly" for those cases where you think developers would better served by not using your library.

The reason it's important to know if adopting a library always carries some level of risk, especially for a library as complex as a text editor. As a potential user of remirror, I'm also a bit concerned as it entirely wraps PM (and vendors it) which means I'm now fully dependent on this library. In other projects that build on a foundation library, it's typical to be able to "drop down" and access the core project. Since Remirror re-exports all of PM, it seems like I'd be interfacing with PM through Remirror which gives me pause. I'm left wondering -- would I ever run into roadblocks that I could solve by interfacing with PM directly? Will I regret the choice to save hours? at the beginning only to be stuck months down the road as I'm locked into Remirror and locked out of the larger PM community? I'd love to hear more about why it would be better to do the work to learn this library instead of doing the work to just learn PM directly and integrate it.

So, in short, if I'm not using the React component (and just using the DOM one), what advantage is there to using this library over just using ProseMirror directly? In what cases would you recommend people not use Remirror and instead use PM?

Thanks!!!

**Comment by ifiokjr** [comment, marked as answer]
Good question, the docs right now aren't good enough, and you're right there's nowhere I make a case for why using `remirror` is preferred.

I'll answer in more depth and add something to the documentation, at a later point but for now here's my quick response.

I find things to be much simpler when similar concerns are close together. That's why the `Extension` based structure of `remirror` works really well for me. You can add a bold extension which takes care of the commands, input rule support, paste matching support, checking whether your current selection has bold **active**, any relevant plugins, all managed for you. With pure ProseMirror each of these things would need custom implementation and managing that can be tedious especially as the editor functionality grows. It can quickly become very messy, even for big companies, without a lot of oversight. 

I hope that `remirror` is that oversight mechanism which allows users to focus their development efforts on higher level features and not low-level structure.

**Reply by scottmessinger** [reply]
Thanks for the response! That's helpful!
